plugins/org.python.pydev/src/org/python/pydev/ui/dialogs/PyDialogHelpers.java

```diff
@@ -161,6 +175,65 @@ public static int openQuestionConfigureInterpreter(AbstractInterpreterManager m)
         return INTERPRETER_CANCEL_CONFIG;
     }
 
+    /**
+     * <p>Statically open a list dialog to choose an interpreter from Conda.</p>
+     * <p>It returns both the interpreter name and executable.</p>
+     * 
+     * @return NameAndExecutable
+     */
+    public static NameAndExecutable openCondaInterpreterSelection() {
+        List<File> envs = CondaPackageManager.listCondaEnvironments(PyDevCondaPreferences.getExecutable());
+        List<NameAndExecutable> treatedEnvs = null;
+
+        if (PlatformUtils.isWindowsPlatform()) {
+            treatedEnvs = getTreatedEnvs(envs, "/python.exe");
+        } else {
+            treatedEnvs = getTreatedEnvs(envs, "/bin/python");
+        }
+
+        String title = "Conda interpreter selection";
+        String message = "Select an intepreter from the list.";
+        Shell shell = EditorUtils.getShell();
+        LabelProvider labelProvider = new LabelProvider() {
+            @Override
+            public Image getImage(Object element) {
+                return ImageCache.asImage(SharedUiPlugin.getImageCache().get(UIConstants.PY_INTERPRETER_ICON));
+            }
+
+            @Override
+            public String getText(Object element) {
+                if (element != null && element instanceof NameAndExecutable) {
+                    NameAndExecutable nameAndExecutable = (NameAndExecutable) element;
+                    return nameAndExecutable.o1;
+                }
+                return super.getText(element);
+            }
+        };
+
+        ListDialog d = new ListDialog(shell);
+        d.setInput(treatedEnvs);
+        d.setContentProvider(new ListContentProvider());
+        d.setLabelProvider(labelProvider);
+        d.setMessage(message);
+        d.setTitle(title);
+        d.open();
+        if (d != null) {
+            return (NameAndExecutable) d.getResult()[0];
+        }
+        return null;
+    }
+
+    private static List<NameAndExecutable> getTreatedEnvs(List<File> envs, String strAddOn) {
+        List<NameAndExecutable> ret = new ArrayList<NameAndExecutable>();
+        for (File env : envs) {
+            String execPath = env.getPath() + strAddOn;
```

Comment:
You shouldn't use string manipulation for dealing with paths.

Use: `new File(env, "python.exe")` to check on windows and `new File(new File(env, "bin"), "python")` or through the `Paths` api -- something as: `Paths.get(env, "bin", "python")`.